packages/doenetml-worker-rust/lib-doenetml-core/src/core/state/state_var.rs

```diff
@@ -678,17 +751,21 @@ impl<T: Default + Clone> StateVar<T> {
     /// Convenience function to call `calculate_state_var_from_dependencies` on interface
     /// and then call mark_fresh
     pub fn calculate_state_var_from_dependencies_and_mark_fresh(&self) {
-        self.interface
-            .calculate_state_var_from_dependencies(&self.value);
+        match self.interface.calculate_state_var_from_dependencies() {
+            StateVarCalcResult::Calculated(val) => self.value.set_value(val),
```

Comment:
This worries me. Do you know for a fact that `came_from_default == False` on `self.value`?

Reply:
If the `calculate` function returns the `Calculated` variant of the enum, that means we are to set `came_from_default` to false. If the `calculate` function returns the `FromDefault` variant of the enum, that means we are to set `came_from_default` to true. We don't have a way to know that the `calculate` function didn't lie when chosing the variant to return, but we act on what it outputs.

Reply:
What I was asking is if it was initialized correctly. I would rather see `self.value.set_value_and_set_came_from_default(val, false)`

Reply:
Makes sense! How about `set_value_from_default` for the function name?

Reply:
Nice. Glad to have help coming up with less awkward names.